hera_cal/vis_clean.py

```diff
@@ -786,6 +787,76 @@ def fft_data(data, delta_bin, wgts=None, axis=-1, window='none', alpha=0.2, edge
     return data, fourier_axes
 
 
+def trim_model(clean_model, clean_resid, dnu, keys=None, noise_thresh=5.0, delay_cut=3000,
+               kernel_size=None, edgecut_low=0, edgecut_hi=0,):
+    """
+    Truncate CLEAN model components below some amplitude threshold.
+
+    Estimate the noise in Fourier space and truncate CLEAN model
+    components below a specified value times the noise.
+
+    Args:
+        clean_model : DataContainer
+            Holds clean_model output of self.vis_clean
+        clean_resid : DataContainer
+            Holds clean_resid output of self.vis_clean
+        dnu : float
+            Frequency channel width [Hz]
+        keys : list of antpairpol tuples
+            List of keys to operate on
+        noise_thresh : float
+            Coefficient times noise to truncate model components below
+        delay_cut : float
+            Minimum |delay| [ns] above which to use in estimating noise
+        kernel_size : int
+            Time median filter kernel_size. None is no median filter.
+        edgecut_low : int
+            Edgecut bins to apply to low edge of frequency band
+        edgecut_hi : int
+            Edgecut bins to apply to high edge of frequency band
+
+    Returns:
+        model : DataContainer
+            Truncated clean_model
+        noise : DataContainer
+            Per integration noise estimate from clean_resid
+    """
+    # get keys
+    if keys is None:
+        keys = [k for k in sorted(set(list(clean_model.keys()) + list(clean_resid.keys()))) if k in clean_model and k in clean_resid]
+
+    # estimate noise in Fourier space
+    model = DataContainer({})
+    noise = DataContainer({})    
+    for k in keys:
+        # get rfft
+        rfft, delays = fft_data(clean_resid[k], dnu, axis=1, window='none', edgecut_low=edgecut_low, edgecut_hi=edgecut_hi, ifft=False, ifftshift=False, fftshift=False)
+        delays *= 1e9
+
+        # get clean_resid NEB
+        neb = noise_eq_bandwidth(~np.isclose(clean_resid[k], 0.0))[:, None]
```

Comment:
I don't understand this step. We're talking about a noisy quantity and we're just checking to see if it's not close to 0? But shouldn't it be averaged on zero? So sometimes it will be... does that matter? Should there be an np.abs() in here? Wouldn't it make more sense to look at the flags?

At the very least, explain what you're doing in the comments.

Reply:
yeah sorry, this is just getting the NEB of a tophat window with flags. so 1 unflagged and 0 flagged. But I'll make this more clear